### PR TITLE
chore: Formatted cell counts with greater precision

### PR DESCRIPTION
**Pull request by atarashansky**
- #3272

### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

### CODE REVIEW

File: frontend/src/views/WheresMyGene/components/HeatMap/utils.ts [hunk 1]
```diff
@@ -390,7 +390,11 @@ export function createYAxisOptions({
             const { total_count } = deserializeCellTypeMetadata(
               value as CellTypeMetadata
             );
-            return total_count > 10000 ? ">10k" : `${total_count}`;
+            let formattedCount = `${total_count}`;
+            if (total_count > 10000) {
+              formattedCount = `${Math.round(total_count / 1000)}k`;
+            }
+            return formattedCount;
```
Comment (seve) [inline frontend/src/views/WheresMyGene/components/HeatMap/utils.ts:397]: const formattedString = Intl.NumberFormat("en-US", {
      maximumFractionDigits: 1,
      notation: "compact",
    }).format(n);
    return `${formattedString}${formattedString !== n.toString() ? "+" : ""}`;
    
    for when we hit 1M of a single cell type 🚀, will also give us  `x.xK`

**Review by seve — CHANGES_REQUESTED** [review]

File: frontend/src/views/WheresMyGene/components/HeatMap/utils.ts (hunk 1 above)
Comment (atarashansky) [reply frontend/src/views/WheresMyGene/components/HeatMap/utils.ts:397]: Oh right, will fix!

**Review by atolopko-czi — APPROVED** [review]
LGTM, with minor Q's:
* Test update needed?
* Is it ok if values [1000..10000) are abbreviated with `k` suffix?

**Review by seve — APPROVED** [review]

**Comment by atarashansky** [comment]
> LGTM, with minor Q's:
> 
> * Test update needed?
I couldn't find tests for the cell count in the tooltip. I'm not sure if they exist.
> * Is it ok if values [1000..10000) are abbreviated with `k` suffix?
The point of the suffix initially was so that the counts do not occupy too much horizontal space. Any number below 10,000 is okay to display without abbreviation.

**Pull request merged**